### PR TITLE
Machines promoted to working

### PR DESCRIPTION
**Pull request by HappyASR**
----------------------------
Mahjong Ougon no Hai (bootleg) [XingXing]

New working clones
------------------
Mahjong Ougon no Hai (bootleg b) [XingXing, little0]

New clones marked as NOT_WORKING
--------------------------------
Mahjong Ougon no Hai (bootleg c) [XingXing, little0]

### CODE REVIEW

File: src/mame/drivers/dynax.cpp [hunk 1]
```diff
@@ -1093,8 +1095,12 @@ uint8_t dynax_state::tenkai_ip_r(offs_t offset)
 
 				// player 1
 				case 0x82:
-					if (m_keyb >= 5)
+					if (m_input_mode != 0xFF)
+						return 0x00;
+					if (m_keyb >= 5){
 						logerror("%04x: unmapped keyb=%02x read\n", m_maincpu->pc(), m_keyb);
+						return 0x00;
+					}
```
Comment (cuavas) [inline src/mame/drivers/dynax.cpp:1103]: Please follow the style of the surrounding code.  Use Allman brace positioning and lowercase letters in hexadecimal literals.

File: src/mame/drivers/dynax.cpp [hunk 2]
```diff
@@ -4909,6 +4923,12 @@ void dynax_state::mjreach(machine_config &config)
 	m_mainlatch->q_out_cb<0>().set(FUNC(dynax_state::flipscreen_w)); // not inverted
 }
 
+void dynax_state::ougonhai(machine_config &config)
+{
+	tenkai(config);
+	tmp91640_device* tmp = reinterpret_cast<tmp91640_device *>(m_maincpu.target());
```
Comment (cuavas) [inline src/mame/drivers/dynax.cpp:4929]: Please use `downcast<...>(*m_maincpu)` – a `reinterpret_cast` may not do the right thing.

File: src/mame/drivers/dynax.cpp [hunk 3]
```diff
@@ -7359,8 +7407,10 @@ GAME( 1991, tenkai2b, tenkai,   tenkai,   tenkai,   dynax_state, empty_init,
...
-GAME( 1991, ougonhaib,ougonhai, tenkai,   tenkai,   dynax_state, empty_init,    ROT0,   "bootleg",                  "Mahjong Ougon no Hai (bootleg)",                                MACHINE_NOT_WORKING | MACHINE_SUPPORTS_SAVE )
+GAME( 1991, ougonhai, 0,        ougonhai,   tenkai,   dynax_state, empty_init,    ROT0,   "Dynax",                    "Mahjong Ougon no Hai",                                          MACHINE_NOT_WORKING | MACHINE_SUPPORTS_SAVE )
+GAME( 1991, ougonhaib,ougonhai, ougonhai,   tenkai,   dynax_state, empty_init,    ROT0,   "bootleg",                  "Mahjong Ougon no Hai (bootleg)",                                MACHINE_SUPPORTS_SAVE )
+GAME( 1991, ougonhaibb,ougonhai, ougonhai,   tenkai,   dynax_state, empty_init,    ROT0,   "bootleg",                  "Mahjong Ougon no Hai (bootleg b)",                                MACHINE_SUPPORTS_SAVE )
+GAME( 1991, ougonhaicb,ougonhai, ougonhai,   tenkai,   dynax_state, empty_init,    ROT0,   "bootleg",                  "Mahjong Ougon no Hai (bootleg c)",                                MACHINE_NOT_WORKING | MACHINE_SUPPORTS_SAVE )
```
Comment (cuavas) [inline src/mame/drivers/dynax.cpp:7413]: Please follow conventions for naming – make the short name `ougonhaib1`, `ougonhaib2` and `ogounhaib3` and “(bootleg set 1)”, “(bootleg set 2)” and “(bootleg set 3)” in the descriptions.

**Review by cuavas — CHANGES_REQUESTED** [review]

**Comment by HappyASR** [comment]
https://anonfiles.com/P6adS6Xbu8/dynax_7z

**Pull request merged**